File: src/main/java/com/example/serveroptvisualizer/concert/ConcertService.java
package com.example.serveroptvisualizer.concert;

import java.util.List;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@Transactional(readOnly = true)
public class ConcertService {

  private final ConcertRepository concertRepository;

  public ConcertService(ConcertRepository concertRepository) {
    this.concertRepository = concertRepository;
  }

  public List<ConcertSummaryView> getConcerts() {
    return concertRepository.findAllByOrderByIdAsc().stream()
        .map(
            concert ->
                new ConcertSummaryView(
                    concert.getId(),
                    concert.getTitle(),
                    concert.getVenue(),
                    concert.getDescription()))
        .toList();
  }
}

File: src/main/kotlin/com/example/serveroptvisualizer/concert/ConcertService.kt
package com.example.serveroptvisualizer.concert

import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional(readOnly = true)
class ConcertService(private val concertRepository: ConcertRepository) {

  fun getConcerts(): List<ConcertSummaryView> =
    concertRepository.findAllByOrderByIdAsc().map { concert ->
      ConcertSummaryView(
        id = concert.id,
        title = concert.title,
        venue = concert.venue,
        description = concert.description,
      )
    }
}
